include/mbgl/storage/default_file_source.hpp

```diff
@@ -2,9 +2,135 @@
 #define MBGL_STORAGE_DEFAULT_FILE_SOURCE
 
 #include <mbgl/storage/file_source.hpp>
+#include <mbgl/util/geo.hpp>
+
+#include <vector>
 
 namespace mbgl {
 
+/*
+ * An offline region defined by a style URL, geographic bounding box, zoom range, and
+ * device pixel ratio.
+ */
+class OfflineTilePyramidRegionDefinition {
+public:
+    OfflineTilePyramidRegionDefinition(const std::string&, const LatLngBounds&, double, double, float);
+
+    std::string styleURL;
```

Comment:
Why is styleUrl and pixelRatio part of this? Isn't the following use case more likely with vector data set?

- createOfflineRegion
- Now apply different styles (with possibly different sprites) to the same offline region.